src/pages/workspaces/workspace/workspace-utils.ts

```diff
@@ -0,0 +1,43 @@
+export type CloudProviderType = 'AZURE' | 'GCP'
+export const cloudProviderTypes: Record<CloudProviderType, CloudProviderType> = {
+  AZURE: 'AZURE',
+  GCP: 'GCP'
+}
+export const isCloudProvider = (x: unknown): x is CloudProviderType => {
+  return x as string in cloudProviderTypes
+}
+
+export interface BaseWorkspaceInfo {
```

Comment:
I think "Info" in names doesn't convey any useful information. I think I'd suggest naming these types `Workspace` and `WorkspaceDetails` to match the API endpoint: https://rawls.dsde-prod.broadinstitute.org/#/workspaces/listWorkspaceDetails